rules/ruleset.go

```diff
@@ -276,9 +246,32 @@ func (as *activeRuleSet) mappingsForRollupID(
 	return policy.DefaultStagedPolicies, false
 }
 
+func (as *activeRuleSet) mappingsForNonRollupIDReverse(id []byte, timeNanos int64,
+	mt metric.Type, at policy.AggregationType) (policy.StagedPolicies, bool) {
+	policies, cutoverNanos := as.mappingsForNonRollupID(id, timeNanos)
+	// NB(cw) aggregation types filter must be applied after the policy list is resolved.
+	// Because policies like [1m:40h|P90,P99; 1m:20h|P50] will be resolved to [1m:40h|P50,P90,P99].
+	filtered, isDefault := filterPoliciesWithAggregationTypes(policies, mt, at, as.pOpts)
+	if isDefault {
```

Comment:
We should only return default staged policies if `cutoverNanos == 0 && isDefault`. So this should be written as:
```
if cutover == 0 && isDefault {
  return policy.DefaultStagedPolicies, true
}
if len(filtered) == 0 && !isDefault {
  return policy.DefaultStagedPolicies, false
}
return policy.NewStagedPolicies(cutoverNanos, false, filtered), true
```

Reply:
Right, sounds good.